resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java

```diff
@@ -167,7 +169,7 @@ public DnsServerAddressStream nameServerAddressStream(String hostname) {
                 if (addresses != null) {
                     return addresses.stream();
                 }
-                return DEFAULT_PROVIDER.nameServerAddressStream(originalHostname);
+                return UNIX_PROVIDER.nameServerAddressStream(originalHostname);
```

Comment:
IIUC, there is a chance that users will never hit this path. Therefore, there is no need to load and initialize `DnsServerAddressStreamProviders` in advance. Can we use `DnsServerAddressStreamProviders.unixDefault()` here instead of a static constant?

Reply:
sure why not...